async.go

```diff
@@ -63,25 +63,36 @@ func (sr *snowflakeRestful) getAsync(
 	defer close(errChannel)
 	token, _, _ := sr.TokenAccessor.GetTokens()
 	headers[headerAuthorizationKey] = fmt.Sprintf(headerSnowflakeToken, token)
-	resp, err := sr.FuncGet(ctx, sr, URL, headers, timeout)
-	if err != nil {
-		logger.WithContext(ctx).Errorf("failed to get response. err: %v", err)
-		sfError.Message = err.Error()
-		errChannel <- sfError
-		return err
-	}
-	if resp.Body != nil {
-		defer resp.Body.Close()
-	}
 
-	respd := execResponse{}
-	err = json.NewDecoder(resp.Body).Decode(&respd)
-	resp.Body.Close()
-	if err != nil {
-		logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
-		sfError.Message = err.Error()
-		errChannel <- sfError
-		return err
+	var err error
+	var respd execResponse
+
+	for {
```

Comment:
Are we sure, that we don't need another condition (like based on time?) here? If something goes wrong we can jump into infinite loop and block the thread forever?

Reply:
I think this is sufficient. It's similar to how [non async queries in progress](https://github.com/snowflakedb/gosnowflake/blob/master/restful.go#L241) are handled. Can you think of any scenario where it might get stuck? Error codes would be handled or else another response code would be returned and it would break out.

Reply:
Ok, if you are sure that it's ok - I wouldn't argue :)